test/transforms/test_add_metapaths.py

```diff
@@ -87,3 +87,46 @@ def test_add_metapaths_max_sample():
 
     meta = transform(data)
     assert meta['metapath_0'].edge_index.size(1) < 9
+
+
+def test_add_weighted_metapaths():
+    torch.manual_seed(12345)
+
+    data = HeteroData()
+    data['a'].num_nodes = 2
+    data['b'].num_nodes = 3
+    data['c'].num_nodes = 2
+    data['d'].num_nodes = 2
+    data['a', 'b'].edge_index = tensor([[0, 1, 1], [0, 1, 2]])
+    data['b', 'a'].edge_index = data['a', 'b'].edge_index.flip([0])
+    data['b', 'c'].edge_index = tensor([[0, 1, 2], [0, 1, 1]])
+    data['c', 'b'].edge_index = data['b', 'c'].edge_index.flip([0])
+    data['c', 'd'].edge_index = tensor([[0, 1], [0, 0]])
+    data['d', 'c'].edge_index = data['c', 'd'].edge_index.flip([0])
+
+    metapaths = [
+        [('a', 'b'), ('b', 'c')],
+        [('a', 'b'), ('b', 'c'), ('c', 'd')],
+        [('a', 'b'), ('b', 'c'), ('c', 'd'), ('d', 'c'), ('c', 'b'),
+         ('b', 'a')],
+    ]
+    transform = AddMetaPaths(metapaths, weighted=True)
+    metapath_data = transform(copy.copy(data))
+
+    # Make sure manually added metapaths compute the correct number of edges
+    assert metapath_data['a', 'c'].edge_weight.tolist() == [1, 2]
+    assert metapath_data['a', 'd'].edge_weight.tolist() == [1, 2]
+    assert metapath_data['a', 'a'].edge_weight.tolist() == [1, 2, 2, 4]
+
+    # Compute intra-table metapaths efficiently
+    metapaths = [[('a', 'b'), ('b', 'c'), ('c', 'd')]]
+    metapath_data = AddMetaPaths(metapaths, weighted=True)(copy.copy(data))
+    metapath_data['d',
+                  'a'].edge_index = metapath_data['a',
+                                                  'd'].edge_index.flip([0])
+    metapath_data['d', 'a'].edge_weight = metapath_data['a', 'd'].edge_weight
+    metapaths = [[('a', 'd'), ('d', 'a')]]
+    metapath_data = AddMetaPaths(metapaths, weighted=True)(metapath_data)
+    del metapath_data['a', 'd']
```

Comment:
Can be dropped?

Reply:
I have this here because this is similar to what we do for `add_intra_table_metapaths` in the order that we do it.  The deletes aren't strictly necessary but I think it'd be good to keep them.